fluentd/out_syslog.rb

```diff
@@ -99,7 +99,7 @@ def emit(tag, es, chain)
         end
         @packet.time = time
         @packet.tag = @random_string
-        if @tag_key.any? && record.any?
+        if !@tag_key.nil? && @tag_key.any? && record.any?
```

Comment:
`if !@tag_key.nil? && @tag_key.respond_to?(:any?) && record.respond_to?(:any?)`
You have to use `respond_to?` to see if an object provides a method.  You cannot use `any?` if the object is not a string:
```
irb(main):001:0> ''.any?
NoMethodError: undefined method `any?' for "":String
	from (irb):1
	from /usr/bin/irb:11:in `<main>'
```

Reply:
Ah - I see from the test that you expect the code to raise an exception.  I think we should not raise exceptions in this code.
`@tag_key` is defined as an array.  To ensure that it is always an array, do the following:
```
 config_param :tag_key, :array, default: []
```
Why are you using `any?` here?  Are you trying to see if the array is not empty and the record is not empty?  Then just use `empty?`:
```
         if !@tag_key.empty? && !record.empty?

Reply:
Thank you, @richm !!  I've pushed the new patch as you suggested.